libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1229,6 +1229,29 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 						utils().leftShiftNumberOnStack(224);
 						return false;
 					}
+	// Another special case for `address(this).balance`. Post-Istanbul, we can use the selfbalance
+	// opcode.
+	// TODO also add optimizer rule `balance(address) -> selfbalance`.
```

Comment:
This needs information about the dialect together with the rules (or applicability function?).

Reply:
Created https://github.com/ethereum/solidity/issues/7542